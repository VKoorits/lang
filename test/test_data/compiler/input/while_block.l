var fib_sum
var fib1 = 1, fib2 = 1

var n = 10

var i = 2
while i < n:
	fib_sum = fib1 + fib2
	fib1 = fib2
	fib2 = fib_sum
	i = i+1
