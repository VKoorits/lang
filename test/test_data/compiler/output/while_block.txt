CONSTANTS:
	1: 7
	2: 2
	3: 1
CODE:
1:	NEW_VAR 1
2:	PUSH 1
3:	PUSH_CONST 1
4:	STORE
5:	PUSH 1
6:	PUSH_CONST 2
7:	MORE
8:	JUMP_IF_NOT 0x68
9:	NEW_VAR 0
10:	PUSH 1
11:	PUSH 1
12:	PUSH_CONST 3
13:	BINARY_SUB
14:	STORE
15:	DELETE_VAR 0
16:	GOTO 0x20
17:	DELETE_VAR 1
18:	EXIT
