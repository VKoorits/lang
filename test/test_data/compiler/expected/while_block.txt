CONSTANTS:
	1: 1
	2: 10
	3: 2
CODE:
1:	NEW_VAR 5
2:	PUSH 2
3:	PUSH_CONST 1
4:	STORE
5:	PUSH 3
6:	PUSH_CONST 1
7:	STORE
8:	PUSH 4
9:	PUSH_CONST 2
10:	STORE
11:	PUSH 5
12:	PUSH_CONST 3
13:	STORE
14:	PUSH 5
15:	PUSH 4
16:	LESS
17:	JUMP_IF_NOT 0x140
18:	NEW_VAR 0
19:	PUSH 1
20:	PUSH 2
21:	PUSH 3
22:	BINARY_ADD
23:	STORE
24:	PUSH 2
25:	PUSH 3
26:	STORE
27:	PUSH 3
28:	PUSH 1
29:	STORE
30:	PUSH 5
31:	PUSH 5
32:	PUSH_CONST 1
33:	BINARY_ADD
34:	STORE
35:	DELETE_VAR 0
36:	GOTO 0x53
37:	DELETE_VAR 5
38:	EXIT
